### PR TITLE
fixes for acceptance tests

### PR DESCRIPTION
**Pull request by tranma**
WIP

more fixes for acceptance tests. tracking https://github.com/anchor/acceptance-tests/pull/54/

### CODE REVIEW

File: lib/Synchronise/Network/Server.hs [hunk 1]
```diff
@@ -405,6 +405,7 @@ notifyUpdate store datasources ik = do
   recordInitialDocument store ik initial'
 
   where
+    policy = acceptAll
```
Comment (thsutton) [inline lib/Synchronise/Network/Server.hs]: We should probably put the policy into the configuration some how.

File: lib/Synchronise/Network/Server.hs [hunk 2]
```diff
@@ -76,17 +72,18 @@ logName = "Synchronise.Server"
 --
 spawnServer :: Configuration -> Int -> IO ()
 spawnServer cfg n = do
-    noticeM logName "Starting server"
-    _ <- bracket start stop $ \state -> do
+  _ <- bracket start stop $ \state -> do
 
       -- Spawn a server implementing the protocol and some workers
       api      <- spawnServerAPI state
       peasants <- spawnServerWorkers state
```
Comment (christian-marie) [inline lib/Synchronise/Network/Server.hs]: I prefer serfs or peons.

File: lib/Synchronise/Network/Server.hs [hunk 3]
```diff
@@ -76,17 +74,18 @@ logName = "Synchronise.Server"
 --
 spawnServer :: Configuration -> Int -> IO ()
 spawnServer cfg n = do
-    noticeM logName "Starting server"
```
Comment (thsutton) [inline lib/Synchronise/Network/Server.hs]: Why to stop logging?

File: test/api-roundtrip-test.hs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    it "replies to good notify requests with success" $ do
+        let note = ChangeNotification "TestEntity" "TestSource" "item1"
+        result <- runSynchroniseZMQ conn $ enqueueChangeNotification note
+        result `shouldBe` Right ()
+
+    it "replies to invalid requests" $
+        -- Right result <- runSynchroniseZMQ conn $ performRequest InvalidHeader
+        -- result `shouldbe` InvalidResponse
+        pendingWith "This cannot be implemented without changing the interface."
```
Comment (thsutton) [inline test/api-roundtrip-test.hs]: Or just shouting down the ZMQ wire.

File: test/policy-test.hs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+suite :: Spec
+suite = do
+    describe "ignoreConflicts merge policy" $ do
+        let d1 = mkPatch [ Ins [D.OKey "value"] "one"
+                         , Ins [D.OKey "pet"] "cat"]
+        let d2 = mkPatch [ Ins [D.OKey "value"] "two"
+                         , Del [D.OKey "food"] ""]
+
+        it "should ignore all changes which conflict" $ do
```
Comment (thsutton) [inline test/policy-test.hs]: Would be nice if the Patch values here had Names.

File: test/policy-test.hs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            let (d',c) = merge ignoreConflicts d1 d2
+            d' `shouldBe` mkPatch [ Ins [D.OKey "pet"] "cat"
+                                  , Del [D.OKey "food"] ""]
+            c  `shouldBe` map mkRej [ Ins [D.OKey "value"] "one"
+                                    , Ins [D.OKey "value"] "two" ]
+
+    describe "rejectAll merge policy" $ do
```
Comment (thsutton) [inline test/policy-test.hs]: Should probably have additional changes w/ non-conflicting keys.

File: test/policy-test.hs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        let d1 = mkPatch [Ins [D.OKey "value"] "one"]
+        let d2 = mkPatch [Ins [D.OKey "value"] "two"]
+
+        it "should ignore all changes" $ do
+            let (d',_) = merge rejectAll d1 d2
+            d' `shouldBe` emptyPatch
```
Comment (thsutton) [inline test/policy-test.hs]: Check the rejects.

File: test/policy-test.hs (hunk 6 above)
Comment (tranma) [inline test/policy-test.hs]: "pet" and "food" don't conflict.

File: test/policy-test.hs (hunk 5 above)
Comment (tranma) [inline test/policy-test.hs]: Patch Names don't matter for ignoreConflicts, perhaps if we have a test for trustOnlySource we should have names

**Comment by thsutton** [comment]
Looks good with a few small changes to unit tests.

**Comment by tranma** [comment]
mergify so we can move on

**Pull request merged**